src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+enum Shapes {
+  CIRCLE = 'circle',
+  TRIANGLE = 'triangle',
+  RECTANGLE = 'rectangle',
+}
+
+enum Colors {
+  RED = 'red',
+  BLUE = 'blue',
+  GREEN = 'green',
+}
+
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shapes.TRIANGLE;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Invalid data, the value cannot be negative');
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('Incorrect value(s) for triangle');
+    }
+  }
 
-export class Triangle implements Figure {}
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(
+      halfPerimeter *
+        (halfPerimeter - this.a) *
+        (halfPerimeter - this.b) *
+        (halfPerimeter - this.c),
+    );
 
-export class Circle implements Figure {}
+    return Math.floor(triangleArea * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape = Shapes.CIRCLE;
+
+  constructor(
+    public color: Colors,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Invalid data, the radius cannot be negative');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape = Shapes.RECTANGLE;
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Colors,
+    public width: number,
+    public height: number,
+  ) {
+    if (this.height <= 0 || this.width <= 0) {
```

Comment:
you can create function like

```
function checkOnError(condition: boolean, errorMessage: string): void {
  if (condition) {
    throw new Error(errorMessage);
  }
}
```
It is needed because you have the same logic in three places.